fe/fe-core/src/main/java/com/starrocks/journal/bdbje/BDBEnvironment.java

```diff
@@ -95,7 +102,47 @@ public class BDBEnvironment {
     private final String helperHostPort;
     private final boolean isElectable;
 
-    public BDBEnvironment(File envHome, String selfNodeName, String selfNodeHostPort,
+    /**
+     * init & return bdb environment
+     * @param nodeName
+     * @return
+     * @throws JournalException
+     */
+    public static BDBEnvironment initBDBEnvironment(String nodeName) throws JournalException, InterruptedException {
+        // check for port use
+        Pair<String, Integer> selfNode = GlobalStateMgr.getCurrentState().getSelfNode();
+        try {
+            if (NetUtils.isPortUsing(selfNode.first, selfNode.second)) {
+                String errMsg = String.format("edit_log_port %d is already in use. will exit.", selfNode.second);
+                LOG.error(errMsg);
+                throw new JournalException(errMsg);
+            }
+        } catch (IOException e) {
+            String errMsg = String.format("failed to check if %s:%s is used!", selfNode.first, selfNode.second);
+            LOG.error(errMsg, e);
+            JournalException journalException = new JournalException(errMsg);
+            journalException.initCause(e);
+            throw journalException;
+        }
+
+        // constructor
+        String selfNodeHostPort = selfNode.first + ":" + selfNode.second;
+
+        String environmentPath = GlobalStateMgr.getCurrentState().getBdbDir();
```

Comment:
Since it has been decoupled, it feels strange to put the getBdbDir() method in GlobalStateMgr. 
Also it is strange to get this dir from a state